formatter/src/main/java/org/togetherjava/formatter/Formatter.java

```diff
@@ -0,0 +1,319 @@
+                        || lastToken == TokenType.OPEN_BRACES
+                        || lastToken == TokenType.SEMICOLON
+                        || lastToken == TokenType.UNKNOWN)) {
+                    append(result, token);
+
+                    break iteration;
+                } else if (type == TokenType.COMMENT || type == TokenType.SEMICOLON
+                        || type == TokenType.ANNOTATION) {
+                    if (type == TokenType.COMMENT) {
+                        result.append(" ");
+                    }
+
+                    append(result, token);
+
+                    if (forClosed) {
+                        result.append("\n");
+
+                        lastNewLine = true;
+                    }
+
+                    break iteration;
+                } else if ((type == TokenType.OPEN_PARENTHESIS && lastToken != TokenType.IDENTIFIER
+                        && lastToken != TokenType.NOT && lastToken != TokenType.BIGGER
+                        && lastToken != TokenType.OPEN_PARENTHESIS) || lastToken == TokenType.PUBLIC
+                        || lastToken == TokenType.PRIVATE || lastToken == TokenType.PROTECTED
+                        || (lastToken == TokenType.BIGGER && type != TokenType.OPEN_PARENTHESIS)
+                        || lastToken == TokenType.RETURN || lastToken == TokenType.COMMA
+                        || lastToken == TokenType.CATCH
+                        || (lastToken == TokenType.CLOSE_PARENTHESIS
+                                && type != TokenType.CLOSE_PARENTHESIS && type != TokenType.DOT
+                                && type != TokenType.COMMA)
+                        || lastToken == TokenType.PACKAGE) {
+                    result.append(" ");
+                }
+
+                if (!forClosed) {
+                    if (type == TokenType.OPEN_PARENTHESIS) {
+                        forClosingLevel++;
+                    } else if (type == TokenType.CLOSE_PARENTHESIS) {
+                        forClosingLevel--;
+                    }
+
+                    if (forClosingLevel == 0) {
+                        forClosed = true;
+                    }
+                }
+
+                if (type == TokenType.FOR) {
+                    forClosed = false;
+                }
+
+                if (!(NON_SPACE_TOKENS.contains(type) || NON_SPACE_TOKENS.contains(lastToken))) {
+                    result.append(" ");
+                }
+
+                append(result, token);
+            }
+
+            lastToken = type;
+        }
+
+        return new StringBuilder(result.toString().replaceAll("([;}])\n\n", "$1\n"));
+    }
+
+    /**
+     * Puts the needed indentation into a StringBuilder
+     *
+     * @param indentation indentation level
+     * @param sb string builder
+     * @author illuminator3
+     */
+    private void putIndentation(int indentation, StringBuilder sb) {
```

Comment:
I am not sure what this is but there is `String#repeat` and `String#indent` in the stdlib. One of those probably does what you want.